### PR TITLE
JENKINS-29298: Initial support for uno-choice plugin

### PR DESCRIPTION
**Pull request by djaara**
JENKINS-29298: Initial implementation of **Active Choices Plugin** [1] plugin support, please take it as first proposal and platform for further improvements. Will be glad to work on it.

[1] https://wiki.jenkins-ci.org/display/JENKINS/Active+Choices+Plugin

### CODE REVIEW

**Comment by daspilker** [comment]
Please try to keep the size of a pull request smaller, so that it is easier to review in a smaller amount of time. E.g. just add `activeChoiceParam` in a first pull request and then add further features in subsequent PRs. And add documentation.

**Comment by djaara** [comment]
Trimmed down to just one variant - `Active Choice Parameter` with only groovy script support, added documentation.

File: docs/Job-reference.md [hunk 1]
```diff
@@ -5833,6 +5834,24 @@ job('example-2') {
 
 (since 1.26)
 
+### Active Choice Parameter
+
+```groovy
+activeChoiceParam(String paramName, Closure closure = null) {
+    description(String description)
+    visibleItemCount(int visibleItemCount)
```
Comment (daspilker) [inline docs/Job-reference.md]: There is no UI option for this. Why does this need to be configured by the DSL then?

File: docs/Job-reference.md [hunk 2]
```diff
@@ -5833,6 +5834,24 @@ job('example-2') {
...
+    groovyScript {
+      script(String script)
+      fallbackScript(String fallbackScript)
+    }
+}
+```
+
+Defines a Active Choice parameter with groovy script as source of parameter options. Requires the
+[Active Choice Parameter Plugin](https://wiki.jenkins-ci.org/display/JENKINS/Active+Choices+Plugin)
+
```
Comment (daspilker) [inline docs/Job-reference.md]: Please add an example.

File: docs/Job-reference.md (hunk 1 above)
Comment (djaara) [inline docs/Job-reference.md]: Good catch. Removed from DSL. Thanks!

File: docs/Job-reference.md (hunk 2 above)
Comment (djaara) [inline docs/Job-reference.md]: Example added.

**Comment by daspilker** [comment]
OK thanks. I squashed your commits and merged them manually.

**Pull request closed without merging**